proof-systems/src/darlin/poly-commit/src/domain_extended/mod.rs

```diff
@@ -190,4 +232,50 @@ impl<G: Group, PC: PolynomialCommitment<G, Commitment = G>> PolynomialCommitment
     fn challenge_to_scalar(chal: Vec<bool>) -> Result<G::ScalarField, Self::Error> {
         PC::challenge_to_scalar(chal)
     }
+
+    #[cfg(feature = "circuit-friendly")]
```

Comment:
I wonder if also the primitive can benefit from the re-ordering (the complexity of sorting should be much smaller than the complexity of doing additional arithmetic operations caused by unsorted commitment vecs).
If so, we could re-order a priori, avoid using the circuit-friendly flag, thus reducing code-complexity a bit